src/main/java/net/minecraftforge/common/ForgeMod.java

```diff
@@ -233,6 +244,67 @@ public String getName()
         }
     }
 
+    public static class BlockModels extends ModelProvider<BlockModelBuilder> {
+        public ModelFile.GeneratedModelFile<BlockModelBuilder> acaciaFenceGate;
```

Comment:
This API is quite clunky. Maybe we should consider creating models during the blockstate creation, like recipe generators create their advancements?

Reply:
I've pushed a rough implementation of combined block state and block model generators to https://github.com/malte0811/MinecraftForge/tree/1.14.x-blockstategen-temp. It might help for blocks that use a different model for each state, but in general it doesn't really improve the API for blocks that reuse the same model for multiple states (which is probably most nontrivial (= at least 2 states) blocks). I have no idea how one could improve it in that case, but I don't have much experience in API design, so that doesn't mean much.

Reply:
I've pushed the version from the temp branch to this branch now, since you seem to prefer it over the old one (guessing from the Discord comment). Does "builders" in that comment only refer to `VariantBlockstate.Builder` or some others as well (e.g. `PartialBlockstate,` which is quite "builder-like")?

Reply:
I would prefer to avoid the use of `new` during building, because of vanilla never seems to use it in any of theirs, instead opting for helper methods and static factories. It would also reduce code because we could attach the sub-builder to the current builder instance automatically, removing the need for an actual `build` method.